service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FishingTripServiceBean.java

```diff
@@ -282,54 +280,48 @@ private VesselDetailsDTO getVesselDetailsDTO(VesselTransportMeansEntity vesselTr
         return detailsDTO;
     }
 
-
-    // need confirmation for removal of this method
-    private void enrichWithAssetsModuleDataIfNeeded(VesselDetailsDTO vesselDetailsDTO) {
-        if (vesselDetailsDTO != null && vesselDetailsDTO.hasEmptyIdentifierValues()) {
-            try {
-                Set<AssetIdentifierDto> vesselIdentifiers = vesselDetailsDTO.getVesselIdentifiers();
-                List<AssetListCriteriaPair> assetListCriteriaPairs = BaseMapper.mapToAssetListCriteriaPairList(vesselIdentifiers);
-                AssetListCriteria criteria = new AssetListCriteria();
-                criteria.getCriterias().addAll(assetListCriteriaPairs);
-                AssetListQuery query = new AssetListQuery();
-                query.setAssetSearchCriteria(criteria);
-                List<Asset> assetList = assetModuleService.getAssetListResponse(query);
-                vesselDetailsDTO.enrichIdentifiers(assetList.get(0));
-            } catch (ServiceException e) {
-                log.error("Error while trying to send message to Assets module.", e);
-            }
-        }
-    }
-
-
     //To process MDR code list and compare with  database:vesselTransportMeansDao and then enrich with asset module
     private void getMdrCodesEnrichWithAssetsModuleDataIfNeeded(VesselDetailsDTO vesselDetailsDTO) {
```

Comment:
Maybe rewrite this method with some early-out if-clauses instead? Such as "if (vesselDetailsDTO == null) { return; }".
It is badly nested...

Reply:
Yep, shortened